stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -446,10 +446,10 @@ public func +<
     S : SequenceType
     where S.Generator.Element == C.Generator.Element
 >(lhs: C, rhs: S) -> C {
-  var lhs = lhs
-  // FIXME: what if lhs is a reference type?  This will mutate it.
-  lhs.appendContentsOf(rhs)
-  return lhs
+  var result = lhs is AnyObject ? C(lhs) : lhs
```

Comment:
A thing does not need to be AnyObject to have reference semantics -- for example, a struct wrapping a class also has reference semantics.